src/generator.cs

```diff
@@ -378,7 +378,7 @@ public GeneratedType (Type t, GeneratedTypes root)
 			// protected against a StackOverflowException - bug #19751
 			// it does not protect against large cycles (but good against copy/paste errors)
 			if (Parent == Type)
-				throw new BindingException (1030, true, "{0} cannot have [BaseType(typeof({1}))] as it creates a circular dependency", Type, Parent);
+				throw new BindingException (1030, true, bgen.Resources.BI1030, Type, Parent);
```

Comment:
I wonder if we could look them up using reflection (based on the error code) ?

Reply:
I think we definitely could like:
```csharp
internal static string GetErrorMessage (errorCode) {
// reflection stuff yo find the property
}
```
and just change the binding exception to something even lazier for us to write:
```csharp
public BindingException (int error, bool, params object[] content) : this (error, bool, GetErrorMessage (error), content);
```

And just call the new constructor with the error code, that calls the reflection and returns the correct string. then we move to:

```csharp
throw new BindingException (1030, true, Type, Parent); // already has the translated message.
```

Reply:
yup
```
throw new BindingException (1030, true, bgen.Resources.BI1030 ...
```
is just a typo magnet :)

Reply:
Oo cool idea. I might borrow that for Xamarin.Android too. One little complication I can think of for the Xamarin.Android case is that sometimes we have multiple separate strings for a single message code. At the moment, the approach we have is to add suffixes on the message code property name like `XA0000_Files` and `XA0000_Directories`. There might still be a clean way to use reflection for a scenario like that. I'll have to muse on it a bit.

Reply:
Oh, one way to handle the multiple strings scenario would be to allow an optional additional suffix argument in the new override method. That could work nicely. Thanks for the inspiration on this!